db.go

```diff
@@ -282,6 +278,26 @@ func Open(path string, mode os.FileMode, options *Options) (*DB, error) {
 	return db, nil
 }
 
+// readFreelist is called on Open and tx.Check. It assumes there are no
+// concurrenct accesses are being made to the freelist.
+func (db *DB) readFreelist() bool {
```

Comment:
I found the return value of this function a bit tricky to reason about.

It might be cleaner to not return bool and fully separate the concerns of loading the freelist in `readFreelist` from the concerns of flushing the freelist in `db.Open`. E.g.:

`db.Open`:

```
...
if db.readOnly {
	return db, nil
}
hasFreeList := db.meta().freelist != pgidNoFreelist
db.readFreelist()

if !db.NoFreelistSync && !hasFreeList {
...
```
`db.readFreelist`:

```
func (db *DB) readFreelist() {
  b.freelistLoad.Do(func() {
		hasFreeList := db.meta().freelist != pgidNoFreelist
		db.freelist = newFreelist()
		if !hasFreeList {
...
```

And we could even add a `db.hasSyncedFreelist()` function to encapsulate the `db.meta().freelist != pgidNoFreelist` check and clean up the code a bit more.